kyo-core/shared/src/main/scala/kyo/internal/FileImplicits.scala

```diff
@@ -0,0 +1,21 @@
+package kyo.internal
+
+import scala.language.implicitConversions
+import scala.quoted.*
+
+opaque type FileNameWithLine <: String = String
+object FileNameWithLine:
+    def apply(value: String): FileNameWithLine   = value
+    implicit inline def derive: FileNameWithLine = ${ Macros.fileNameWithLine }
```

Comment:
The idiomatic way in Scala 3 would be `inline given derive: FileNameWithLine`

Reply:
Good to know, thanks